### PR TITLE
BE-FE update create order and trading room view for BUY OFFERS

### PR DESCRIPTION
**Pull request by duongquang18it3**
# Ticket
https://github.com/mezonai/mmn-tx-explorer/issues/732

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: dong-service/services/p2p_order_service.go [hunk 1]
```diff
@@ -261,8 +279,15 @@ func (s *OrderService) ConfirmOrderAsSeller(ctx context.Context, orderID int64,
 			return err
 		}
 
+		var isTargetWallet *string
+		if offer.Side == "BUY" {
```
Comment (vietdangquoc) [inline dong-service/services/p2p_order_service.go]: "BUY" using const

File: dong-service/models/p2p_order.go [hunk 2]
```diff
@@ -32,14 +32,12 @@ type Order struct {
 	CreatedAt                 time.Time          `json:"created_at" db:"created_at"`
 	UpdatedAt                 time.Time          `json:"updated_at" db:"updated_at"`
 	BankInfo                  *string            `json:"bank_info,omitempty" db:"-"`
-	OfferCreatorWalletAddress *string            `json:"offer_creator_wallet_address,omitempty" db:"-"`
```
[the remark this answers is not in the document]
Comment (duongquang18it3) [reply dong-service/models/p2p_order.go]: Dạ, bị nhầm phần này. Em đã rollback

File: indexer/internal/storage/postgres.go [hunk 3]
```diff
@@ -2651,3 +2703,391 @@ func (p *PostgresConnector) updateOfferStatus(
...
+
+	if len(releaseOfferIDs) > 0 {
+		if err := p.releaseIntermediaryWalletsIfNoOpenOrders(ctx, tx, releaseOfferIDs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *PostgresConnector) releaseIntermediaryWalletsIfNoOpenOrders(ctx context.Context, tx *sql.Tx, offerIDs []int64) error {
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: bỏ đi

File: indexer/internal/storage/postgres.go [hunk 4]
```diff
@@ -2651,3 +2703,367 @@ func (p *PostgresConnector) updateOfferStatus(
 
 	return nil
 }
+
+func (p *PostgresConnector) updateOrderStatus(
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: where status = waiting transaction
update set status = open
